### PR TITLE
Add Vpc , Vpc peering, Route, Subnet

### PR DESCRIPTION
**Pull request by d-apurva**
This PR adds a capability to manage Huawei's VPC Peering Connection, Routes, Vpc, and Subnets.

### CODE REVIEW

**Comment by niuzhenguo** [comment]
Please resolve conflicts and update CHANGELOG, thanks!

**Comment by frisky-dis** [comment]
Done

File: huaweicloud/resource_huaweicloud_vpc_subnet_v1.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+	n, err := subnets.Create(subnetClient, createOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error creating Huaweicloud VPC subnet: %s", err)
+	}
+
+	d.SetId(n.ID)
+	log.Printf("[INFO] Vpc Subnet ID: %s", n.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"CREATING"},
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_vpc_subnet_v1.go]: According to the API doc and feedbacks from users, "UNKNOWN" should be included as Pending status, and seems there's no a status "CREATING".

File: huaweicloud/resource_huaweicloud_vpc_subnet_v1.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+	config := meta.(*Config)
+	subnetClient, err := config.networkingV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating Huaweicloud networking client: %s", err)
+	}
+	vpc_id := d.Get("vpc_id").(string)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"ACTIVE"},
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_vpc_subnet_v1.go:233]: ditto.

File: huaweicloud/resource_huaweicloud_vpc_subnet_v1.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+			return nil, "", err
+		}
+
+		if n.Status == "ACTIVE" {
+			return n, "ACTIVE", nil
+		}
+
+		//If subnet status is other than Active, send error
+		if n.Status == "DOWN" || n.Status == "ERROR" || n.Status == "UNKNOWN" {
+			return nil, "", fmt.Errorf("Subnet status: '%s'", n.Status)
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_vpc_subnet_v1.go]: UNKNOWN here is a normal status.

File: huaweicloud/resource_huaweicloud_vpc_subnet_v1.go (hunk 2 above)
Comment (frisky-dis) [reply huaweicloud/resource_huaweicloud_vpc_subnet_v1.go:233]: Done

File: huaweicloud/import_huaweicloud_vpc_peering_connection_v2_test.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package huaweicloud
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+// PASS
```
Comment (niuzhenguo) [inline huaweicloud/import_huaweicloud_vpc_peering_connection_v2_test.go]: Please help to clean up such test results "// PASS"

Comment (frisky-dis) [reply huaweicloud/import_huaweicloud_vpc_peering_connection_v2_test.go]: done

**Pull request merged**